src/new-client/src/App/Trades/TradesHeader/ExcelButton.tsx

```diff
@@ -40,10 +44,47 @@ const Button = styled("button")`
     transform: scale(0.7);
   }
 `
+export const [useExportTable, exportTable$] = bind(
+  tableTrades$.pipe(
+    map((trades) =>
+      trades.map((trade) =>
+        colConfigs.map(({ field, numeric, valueFormatter }) => {
+          let res = valueFormatter?.(trade[field]) ?? trade[field]
+          if (typeof res === "string" && res?.includes(",")) {
+            res = '"' + res + '"'
+          }
+          return res
+        }),
+      ),
+    ),
+  ),
+  [],
+)
 
 export const ExcelButton: React.FC = () => {
+  const trades = useExportTable()
```

Comment:
Because this component doesn't render any of this trade data to the UI, we shouldn't actually need this hook, which brings the trades into the component state.  In fact, the whole `download_csv` function (which I would rename to  `downloadCsv` for consistency) should be moveable up and out of the component, since it doesn't use any component state.   So the idea would be:  

1. No need for `bind`, just define `const exportTable$ = tableTrades$.pipe(...)` directly
2. Add the `take(1)` after your `map`
3. Inside `downloadCsv`, do 
```
exportTable$.subscribe((trades) => {
.... your csv code
})
```

Because it's a `take(1)` and all of this happens outside the component, the Observable won't hang around in memory  between clicks.